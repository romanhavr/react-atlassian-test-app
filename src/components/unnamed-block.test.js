// @flow

import { shallow } from 'enzyme';
import React from 'react';
import UnnamedBlock from './unnamed-block';

const initialDispatches = {
    sortTable: jest.fn(),
    userFetchReq: jest.fn()
}

const shallowWrapper = shallow(<UnnamedBlock {...initialDispatches} />);

describe('Unnamed Block component testing...', () => {
    
    const creatIssueButton = shallowWrapper.find('[data-test="create-issue-button"]');
    const sagaActionButton = shallowWrapper.find('[data-test="saga-action-button"]');
    const select = shallowWrapper.find('[data-test="input-select"]');

    it('Unnamed Block should "click" Create Issue Button', () => {
        expect(creatIssueButton.length).toBe(1);
        
        creatIssueButton.simulate('click');
        expect(initialDispatches.sortTable).toHaveBeenLastCalledWith(null);
    })

    it('Unnamed Block should "click" Saga Action Button', () => {
        expect(sagaActionButton.length).toBe(1);

        sagaActionButton.simulate('click', 'SAGA Action');
        expect(initialDispatches.userFetchReq).toHaveBeenLastCalledWith('SAGA Action')
    })

    it('Unnamed Block should "select" new sorting data', () => {
        expect(select.length).toBe(1);

        select.simulate('change', { label: 'Assignee', value: 'assignee' });
        expect(initialDispatches.sortTable).toHaveBeenLastCalledWith({ label: 'Assignee', value: 'assignee' })
    })

    it('Unnamed Block should include SNAPSHOT', () => {
        const tree = shallow(<header></header>);
        expect(tree).toMatchSnapshot();
    })
});