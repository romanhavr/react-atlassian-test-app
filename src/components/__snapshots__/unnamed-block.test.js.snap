// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unnamed Block component testing... Unnamed Block should include SNAPSHOT 1`] = `<header />`;
